nutrition_labels/grant_data_processing.py

```diff
@@ -1,100 +1,190 @@
+
+    # No need to include tags if no grant number is given or you don't want to include
+    # the grant in the training data
+    rf_tags.dropna(subset=['Grant Reference', 'Normalised code'], inplace=True)
+
+    # Create list of dicts for each WT grant number given
+    rf_list = []
+    for i, row in rf_tags.iterrows():
+        # Strip whitespace from grant ref, since sometimes it has it
+        rf_list.append({
+            'RF question':row['file'],
+            'RF Name':row['Name'],
+            'Normalised code - RF':int(row['Normalised code']),
+            'Internal ID':row['Grant Reference'].strip(),
+            })
+
+    rf_df = pd.DataFrame(rf_list)
+    rf_df.drop_duplicates(subset=['Internal ID', 'Normalised code - RF'], inplace=True)
+
+    return rf_df
+
+def process_grants(grant_tags, grants_code_dict):
+
+    # If Nonie and Liz have labelled it use Nonies
+    grant_tags = merge_tags(grant_tags, 'tool relevent ', 'Liz code')
+    grant_tags['Normalised code'] = [grants_code_dict[str(int(code))] for code in grant_tags['Merged code']]
+
+    # No need to include tags if you don't want to include
+    # the grant in the training data
+    grant_tags.dropna(subset=['Normalised code'], inplace=True)
+
+    # Create list of dicts for each WT grant number given
+    grants_list = []
+    for i, row in grant_tags.iterrows():
+        # Strip whitespace from grant ref, since sometimes it has it
+        grants_list.append({
+            'Normalised code - grants':int(row['Normalised code']),
+            'Internal ID':row['Internal ID'],
+            })
+    grants_df = pd.DataFrame(grants_list)
+    grants_df.drop_duplicates(subset=['Internal ID', 'Normalised code - grants'], inplace=True)
+
+    return grants_df
+
+if __name__ == '__main__':
+    
+    # load data
+    epmc_tags_query_one = pd.read_csv('data/raw/EPMC_relevant_tool_pubs_manual_edit_Lizadditions.csv', encoding = "latin")
+    epmc_tags_query_two = pd.read_csv('data/raw/EPMC_relevant_pubs_query2_manual_edit.csv')
+    rf_tags = pd.read_csv('data/raw/ResearchFish/research_fish_manual_edit.csv')
+    grant_tags = pd.read_csv('data/raw/wellcome-grants-awarded-2005-2019_manual_edit_Lizadditions.csv')
+    grant_data = pd.read_csv('data/raw/wellcome-grants-awarded-2005-2019.csv')
+
+    # Normalising the codes so they are all similar for different data sources
+    # 'None' if you dont want to include these tags in the training data
+    epmc_code_dict = {'1': 1, '2': 2, '3': 3, '4': None, '5': None, '6': None}
+    rf_code_dict = {'1': 1, '2': 2, '3': 3, '4': None, '5': None}
+    grants_code_dict = {'1': 1, '4': None, '5': 5}
+
+    # Process each of the 3 data sources separately and output a 
+    # dataframe for each of grant numbers - cleaned tags links
+    epmc_df = process_epmc(epmc_tags_query_one, epmc_tags_query_two, epmc_code_dict)
+    rf_df = process_RF(rf_tags, rf_code_dict)
```

Comment:
this seems to be breaking because of '55' in the codes did you clean them out before hand?

Reply:
oops yes i did in the excel sheet. I'll upload the cleaned versions to sharepoint

Reply:
i've added the cleaned RF data. let me know if it all works when using that (i think the other files should be ok)